qiskit/ignis/experiments/calibration/cal_base_generator.py

```diff
@@ -12,112 +12,75 @@
-
+                 name: str,
+                 qubit: int,
+                 parameters: Dict[str, Union[int, float, complex, circuit.Parameter]],
+                 values_to_scan: Iterable[float],
+                 ref_frequency: Optional[float] = None):
+        """
         Args:
-            cal_name: Name of this calibration.
-            target_qubits: Target qubit of this calibration.
-            cal_generator: Generator of calibration circuit.
-            table: Table of calibration data.
+            qubit: The qubit to which we will add the calibration.
+            parameters: The arguments for the pulse schedule.
+            values_to_scan: A list of amplitudes for which to generate a circuit.
+            ref_frequency: A reference frequency of drive channel to run calibration.
+                Usually this is identical to the qubit frequency, or f_01.
+                The drive channel is initialized with this frequency and
+                shift frequency instruction of consecutive pulses, or pulse sideband,
+                are modulated with respect to this frequency.
         """
-        try:
-            self.target_qubits = list(target_qubits)
-        except TypeError:
-            self.target_qubits = [target_qubits]
-
-        # calibration generator is not agnostic to backend because
-        # pulse schedule is specific to each physical qubit.
-        # this value is used to validation of backend consistency when submitting job.
-        self.associated_backend = table.backend_name
-
-        super().__init__(name=cal_name, qubits=target_qubits)
-
-        self._defined_parameters = set()
-        self._circuits = []
-        self._metadata = []
-
-        cal_prog = cal_generator(
-            name=self.name,
-            table=table,
-            target_qubits=target_qubits,
-            **kwargs
-        )
-
-        # store parameters in generated circuits
-        for temp_circ in cal_prog.circuits:
-            self._defined_parameters.update(temp_circ.parameters)
-
-        # validation
-        if len(cal_prog.circuits) != len(cal_prog.metadata):
-            raise CalExpError(
-                'Number of circuits and metadata are not identical.'
-                '{}!={}'.format(len(cal_prog.circuits), len(cal_prog.metadata)))
-
-        self._unassigned_circuits = cal_prog.circuits
-        self._unassigned_metadata = cal_prog.metadata
+        super().__init__(name=name, qubits=[qubit])
+        self._parameters = parameters
+        self._scanned_values = values_to_scan
+        self._ref_frequency = ref_frequency
 
-    def assign_parameters(self,
-                          parameters: Dict[Union[str, Parameter], Iterable[Union[int, float]]]):
-        """Assign values to scan for parameters.
-
-        If length of bind values are different, scan is limited to the parameter
-        with the shortest length.
+    def _template_qcs(self) -> List[circuit.QuantumCircuit]:
```

Comment:
```suggestion
    def _template_qcs(self) -> List[QuantumCircuit]:
```
Can we use `QuantumCircuit` instead of `circuit.QuantumCircuit` to make the code easier to read?